java/code/src/org/keyczar/KeyczarKey.java

```diff
@@ -57,6 +58,8 @@
   private static final int PBE_SALT_SIZE = 8;
   private static final int IV_SIZE = 16;
   private static final int PBE_ITERATION_COUNT = 1000;
+  
+  protected ArrayList<byte[]> fallbackHash = new ArrayList<byte[]>();
```

Comment:
getFallbackHash should return an empty arraylist for every keytype that didn't have a keyczar version with a buggy implementation, so It's defined here, ideally it shouldn't have to be used much. I think I only implemented it with super call on the public keys so that if it was needed in a future version it was obvious where to add the fallback hash.